### DISCUSSION TITLE
Magic AI

### DISCUSSION THREAD

**Discussion by chrisBradley-dev** [unopim/unopim#72] [Ideas]
I think it would be more efficient to to have a button on the Magic AI that gets all the current AI models and loads them into a list using this call

`get https://api.openai.com/v1/models`

<img width="1402" alt="Screenshot 2024-12-06 at 2 59 55 pm" src="https://github.com/user-attachments/assets/32b5b5b6-bd7e-49e3-a798-92fd78382be1">

Then have a checkbox of models allowed to be used in Magic AI instead of hard coding the AI function. 

<img width="1401" alt="Screenshot 2024-12-06 at 3 01 32 pm" src="https://github.com/user-attachments/assets/c831fcaa-d177-4ccd-913c-5cae7270ad35">

When using Magic AI, it would be great to add contextual data to the call that is already on the product.

<img width="632" alt="Screenshot 2024-12-06 at 3 03 56 pm" src="https://github.com/user-attachments/assets/58edbd9d-9ed9-4da7-9bfb-e05fd2043643">

**Comment by navneetkumar-pim-webkul** [comment]
@chris-chameleonbrands 

Thank you for sharing your idea! Introducing a button in Magic AI to dynamically fetch and display AI models using the GET https://api.openai.com/v1/models endpoint is a clever approach. Allowing users to select models via checkboxes instead of hardcoding the functionality would definitely make the system more flexible and user-friendly.

We also appreciate your suggestion to include contextual product data in Magic AI calls. This could enhance the relevance and accuracy of responses, tailoring the AI behavior more closely to the user's needs.

Currently, we provide the option to select the AI model on the product page, but we’ll explore ways to make this process even more intuitive and user-friendly.

We'll review the feasibility of these features and consider them for future updates. 


Thanks

**Reply by chrisBradley-dev** [reply]
@navneetbhardwaj thanks, I think it would greatly enhance usage of the Magic AI component and reduce UnoPim's requirement to continuously update the list of models with subsequent releases.
